commands/mute/tempmute.js

```diff
@@ -0,0 +1,127 @@
+                }
+            }
+    
+            // Adds Muted role to user.
+            toTempMute.roles.add(msg.guild.roles.cache.find(role => role.name === "Muted"));
+            msg.channel.send(`${toTempMute} was muted by ${msg.member}.\nReason: ${reason}`);
+
+            // Sends user a DM notifying them of their muted status.
+            toTempMute.send("You've been muted for " + lengthOfTime + " for the following reason: ```" + reason + " ```");
+        }
+
+        // Outputs a message to the audit-logs channel
+        let channel = msg.guild.channels.cache.find(channel => channel.name === 'audit-logs')
+
+        const tempmuteEmbed = new Discord.MessageEmbed()
+            .setColor('#0099ff')
+            .setTitle(`${toTempMute.user.username}#${toTempMute.user.discriminator} was muted by ${msg.author.tag}:`)
+            .setDescription(reason)
+            .setThumbnail(`https://cdn.discordapp.com/avatars/${toTempMute.user.id}/${toTempMute.user.avatar}.png`)
+            .setTimestamp()
+            .setFooter(`${msg.guild.name}`);
+
+            channel.send(tempmuteEmbed);
+
+        const tempUnMuteEmbed = new Discord.MessageEmbed()
+            .setColor('#0099ff')
+            .setTitle(`${toTempMute.user.username}#${toTempMute.user.discriminator} was unmuted after ${lengthOfTime}:`)
+            .setThumbnail(`https://cdn.discordapp.com/avatars/${toTempMute.user.id}/${toTempMute.user.avatar}.png`)
+            .setTimestamp()
+            .setFooter(`${msg.guild.name}`);
+
+        // Add record to infractions table.
+        let now = new Date();
+        let timestamp = dateFormat(now, "yyyy-mm-dd HH:MM:ss");
+
+        var sqlInfractions = `INSERT INTO infractions (timestamp, user, action, length_of_time, reason, valid, moderator) 
+        VALUES ('${timestamp}', '${toTempMute}', 'cc!tempmute', '${lengthOfTime}', '${reason}', true, '${msg.author.tag}')`;
+
+        con.query(sqlInfractions, function (err, result) {
+            if (err) {
+            console.log(err);
+            } else {
+            console.log("1 record inserted into infractions.");
+            }
+        });
+
+        var sqlModLog = `INSERT INTO mod_log (timestamp, moderator, action, length_of_time, reason) 
+        VALUES ('${timestamp}', '${msg.author.tag}', '${msg}', '${lengthOfTime}', '${reason}')`;
+
+        con.query(sqlModLog, function (err, result) {
+            if (err) {
+            console.log(err);
+            } else {
+            console.log("1 record inserted into mod_log.");
+            }
+        });
+
+        const unmute = (message) => {
+            toUnmute = message.mentions.members.first();
+            toUnmute.roles.remove(message.guild.roles.cache.find(role => role.name === "Muted"));
+            message.channel.send(`${toUnmute} was unmuted.`);
+            toUnmute.send("You've been unmuted.");
+        }
+
+        timestamp = dateFormat(now, "yyyy-mm-dd HH:MM:ss");
```

Comment:
Won't setting the timestamp outside of the setTimeout function set it to the time that the command was run instead of the time he was unmuted?

![image](https://user-images.githubusercontent.com/26176516/108642765-2246e800-7475-11eb-92cb-9d339dea85ef.png)